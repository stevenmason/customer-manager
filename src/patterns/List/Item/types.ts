interface Props {
  children: React.ReactNode;
}

export default Props;
