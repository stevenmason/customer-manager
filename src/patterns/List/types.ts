interface Props {
  loading?: boolean;
  children: React.ReactNode;
}

export default Props;
