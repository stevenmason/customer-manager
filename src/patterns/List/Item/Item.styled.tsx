import styled from "styled-components";

const StyledItem = styled.div`
  display: flex;
  justify-content: space-between;
  padding: 30px 10px;
  background-color: #f1f1f1;

  &:not(:last-child) {
    border-bottom: 1px solid #d7dbdf;
  }
`;

export default StyledItem;
