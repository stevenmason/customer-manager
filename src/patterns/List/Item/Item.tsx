import React from "react";

import Props from "./types";
import StyledItem from "./Item.styled";

const Item: React.FC<Props> = ({ children }) => (
  <StyledItem>{children}</StyledItem>
);

export default Item;
