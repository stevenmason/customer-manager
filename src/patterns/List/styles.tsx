import styled from "styled-components";

const List = styled.div`
  background-color: white;
  margin-top: 20px;
`;

const InfoText = styled.span`
  display: block;
  text-align: center;
  margin-bottom: 10px;
`;

export { List, InfoText };
