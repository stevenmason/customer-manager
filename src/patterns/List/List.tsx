import React from "react";

import Props from "./types";
import * as SC from "./styles";

const List: React.FC<Props> = ({ loading, children }) => (
  <SC.List>
    <SC.InfoText>
      {loading
        ? "...Loading"
        : `Showing ${
            children instanceof Array ? children.length : 1
          } Customers`}
    </SC.InfoText>
    {children}
  </SC.List>
);

export default List;
