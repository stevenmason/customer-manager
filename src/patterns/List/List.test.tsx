import React from "react";
import renderer from "react-test-renderer";

import List from "./List";
import Item from "./Item";

const PROPS = {};

it("renders correctly", () => {
  const tree = renderer
    .create(
      <List {...PROPS}>
        <Item>Test</Item>
      </List>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
