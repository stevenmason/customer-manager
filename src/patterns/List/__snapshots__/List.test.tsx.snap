// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-bdVaJa jtYfyA"
>
  <span
    className="sc-bwzfXH fDIBjQ"
  >
    Showing 1 Customers
  </span>
  <div
    className="sc-htpNat dISaAK"
  >
    Test
  </div>
</div>
`;
